### PR TITLE
remove retry in TestNetworkUpdateTriggers

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

now it has used channel notification, so we donot need retry anymore

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@hanxiaop Unless I miss the cb will be run multiple times, and then the channel notification is redundant

File: pilot/pkg/serviceregistry/kube/controller/network_test.go [hunk 1]
```diff
@@ -79,12 +78,9 @@ func TestNetworkUpdateTriggers(t *testing.T) {
 	expectGateways := func(t *testing.T, expectedGws int) {
 		// wait for a notification
 		assert.ChannelHasItem(t, notifyCh)
-		retry.UntilSuccessOrFail(t, func() error {
```
Comment (hanxiaop) [inline pilot/pkg/serviceregistry/kube/controller/network_test.go:82]: I tried this, and there were test failures when running it 100 times.

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/network_test.go:82]: Fixed, the reason is that the cb has not finished

**Review by hanxiaop — APPROVED** [review]
LGTM

**Comment by hzxuzhonghu** [comment]
ping @ramaraochavali

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**